src/frontend/src/components/authenticateBox.ts

```diff
@@ -0,0 +1,227 @@
+      userNumber: getUserNumber(),
+    });
+  });
+};
+
+export const authenticateBoxTemplate = (props: PageProps): void => {
+  const container = document.getElementById("pageContent") as HTMLElement;
+  render(pageContent(props), container);
+};
+
+const pageContent = ({
+  templates,
+  addDevice,
+  onContinue,
+  recoverAnchor,
+  register,
+  userNumber,
+}: PageProps): TemplateResult => {
+  const anchorInput = mkAnchorInput({
+    userNumber,
+    onSubmit: onContinue,
+  });
+
+  return html` <div class="l-container c-card c-card--highlight">
+      <!-- The title is hidden but used for accessibility -->
+      <h1 class="is-hidden">Internet Identity</h1>
+      <div class="c-logo">${icLogo}</div>
+      <div class="l-stack t-centered">
+        ${templates.message} ${templates.chasm ? mkChasm(templates.chasm) : ""}
+      </div>
+
+      ${anchorInput.template}
+
+      <button
+        @click="${anchorInput.submit}"
+        id="authorizeButton"
+        class="c-button"
+      >
+        Authorize
+      </button>
+      ${mkLinks({
+        register,
+        recoverAnchor: () => recoverAnchor(anchorInput.readUserNumber()),
+        addDevice,
+      })}
+    </div>
+    ${footer}`;
+};
+
+export const authenticate = async (
+  connection: Connection,
+  userNumber: bigint
+): Promise<LoginFlowSuccess | LoginFlowError> => {
+  try {
+    const result = await withLoader(() => connection.login(userNumber));
+    return apiResultToLoginFlowResult(result);
+  } catch (error) {
+    return {
+      tag: "err",
+      title: "Authentication Failed",
+      message: "An error occurred during authentication.",
+      detail: error instanceof Error ? error.message : JSON.stringify(error),
+    };
+  }
+};
+
+const mkChasm = ({ info, message }: ChasmOpts): TemplateResult => {
+  /* the chasm that opens to reveal details about alternative origin */
+  const chasmRef: Ref<HTMLDivElement> = createRef();
+
+  /* the (purely visual) arrow on the chasm */
+  const chasmToggleRef: Ref<HTMLSpanElement> = createRef();
+
+  /* Toggle the chasm open/closed */
+  const chasmToggle = () =>
+    withRef(chasmRef, (chasm) => {
+      const classes = chasm.classList;
+
+      if (classes.contains("c-chasm--closed")) {
+        classes.remove("c-chasm--closed");
+        classes.add("c-chasm--open");
+
+        withRef(chasmToggleRef, (arrow) =>
+          arrow.classList.add("c-chasm__button--flipped")
+        );
+      } else if (classes.contains("c-chasm--open")) {
+        classes.remove("c-chasm--open");
+        classes.add("c-chasm--closed");
+
+        withRef(chasmToggleRef, (arrow) =>
+          arrow.classList.remove("c-chasm__button--flipped")
+        );
+      }
+    });
+
+  return html`
+    <p class="t-paragraph t-weak"><span id="alternative-origin-chasm-toggle" class="t-action" @click="${chasmToggle}" >${info} <span ${ref(
+    chasmToggleRef
+  )} class="t-link__icon c-chasm__button">${caretDownIcon}</span></span><br/>
```

Comment:
you need the `br`? if the div is `display: block`?

Reply:
(I don't like br, no particular reason)

Reply:
I don't, good call (although this code was just moved around). Removed!